src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -251,6 +252,38 @@ public void shouldUseMainConnectionForSelectFunctionCalls() throws SQLException
             .containsOnly(MAIN);
     }
 
+    @Test
+    public void shouldUseReplicaForKnownReadOnlyFunctionCalls() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+
+        connection.prepareStatement("SELECT count(*) FROM user").executeQuery();
```

Comment:
Could we use the queries that caused the problem in the first place?

Reply:
Red-green-refactor would be nice though

Reply:
> Could we use the queries that caused the problem in the first place?

I don't want to use in-product queries in OS lib. I don't have time and energy to think if I'm not leaking too much.

> Red-green-refactor would be nice though

It's nice but also has drawbacks. With RGR master is not green all the. It makes git bisect harder. You can always rollback non-test changes and do RGR while reviewing the change.